src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala

```diff
@@ -323,23 +321,18 @@ final case class ClientsApiServiceImpl(
   private def decorateKey(
     key: AuthorizationProcessModel.Key
   )(implicit contexts: Seq[(String, String)]): Future[PublicKey] = {
-    val result = for {
-      relationship     <- partyProcessService.getRelationship(key.relationshipId)
-      user             <- userRegistryService.findById(relationship.from)
-      relationshipInfo <- PartyProcessConverter.toApiRelationshipInfo(user, relationship)
-    } yield relationshipInfo.state match {
-      case RelationshipState.ACTIVE => key.toApi(isOrphan = false, user)
-      case _                        =>
-        // It could be that in the new API, when a user is removed, a 404 error is returned.
-        // At this moment, it should return the relationship with the state set to Deleted.
-        key.toApi(isOrphan = true, user)
-    }
-
-    result.recoverWith {
-      case PartyProcessApiError(404, _, _, _, _) =>
-        Future.successful(key.toApi(isOrphan = true, UserResource(id = key.relationshipId)))
-      case other                                 => Future.failed(other)
-    }
+    for {
+      selfcareUuid     <- getSelfcareIdFutureUUID(contexts)
+      userResponse     <- selfcareV2ClientService.getUserById(selfcareUuid, key.userId).recoverWith {
+        case _: UserNotFound => Future.successful(UserResponse())
+      }
+      (user, isOrphan) <- userResponse match {
+        case UserResponse(_, id, _, _, _) if id == None => Future.successful((User(userId = key.userId, "", ""), true))
+        case _                                          =>  for {
+          user <- userResponse.toApi.toFuture
+        } yield (user, false)
+      }
+    } yield key.toApi(user = user, isOrphan = isOrphan)
```

Comment:
I still see 2 behaviours that are different compared to the actual version:
1. if the user is not found on selfcare, the name and surname are set to empty strings
2. if the user is found on selfcare, but any of the fields is missing (eg the name) the call fails

[This conversion](https://github.com/pagopa/interop-be-backend-for-frontend/pull/240/files#diff-331643c325f350b487d2dbe3ba7dfe88a942fcbbfe4ea2fd26a02f4a3d57a3bcR39) is doing more that it should and is not required, we can keep [this logic](https://github.com/pagopa/interop-be-backend-for-frontend/pull/240/files#diff-48f26966b8f07ec6501b901ec1d2b5cdd4a1c1af372e560079a1216179a410ddL97-L109) which handles the necessary fields

Using the old conversion the code could become
```suggestion
      isOrphan = userResponse.id.isEmpty
      user = userResponse.toApi(userId)
    } yield key.toApi(user = user, isOrphan = isOrphan)
```

Does it work for you?